docs/topics/gradle/gradle-compilation-and-caches.md

```diff
@@ -13,11 +13,20 @@ On this page, you can learn about the following topics:
 
 ## Incremental compilation
 
-The Kotlin Gradle plugin supports incremental compilation. Incremental compilation tracks changes to files in the classpath
-between builds so that only the files affected by these changes are compiled. Incremental compilation works with [Gradle's
-build cache](#gradle-build-cache-support) and supports [compilation avoidance](https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_compile_avoidance).
+The Kotlin Gradle plugin supports incremental compilation, which is enabled by default for Kotlin/JVM and Kotlin/JS projects.
+Incremental compilation tracks changes to files in the classpath between builds so that only the files affected
+by these changes are compiled.
+This approach works with [Gradle's build cache](#gradle-build-cache-support) and supports [compilation avoidance](https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_compile_avoidance).
 
-Incremental compilation is supported for Kotlin/JVM and Kotlin/JS projects, and is enabled by default.
+For Kotlin/JVM, incremental compilation relies on classpath snapshots,
+which capture the API structure of modules to determine when recompilation is necessary.
+You can configure classpath snapshots as **fine-grained** or **coarse-grained**, depending on the level of detail required:
```

Comment:
It's not a public option. I'd rather say something like this:

```
To optimize the overall pipeline, Kotlin compiler uses two types of classpath snapshots: fine-grained and coarse-grained. 

<some explanations follow>
```

Reply:
I see - I like this version a lot - updated it!